domain/objectstore/service/service.go

```diff
@@ -64,6 +72,25 @@ func (s *Service) GetMetadata(ctx context.Context, path string) (objectstore.Met
 	}, nil
 }
 
+// GetMetadataBySHA256Prefix returns the persistence metadata for the object
+// with SHA256 starting with the provided prefix.
+func (s *Service) GetMetadataBySHA256Prefix(ctx context.Context, sha256 string) (objectstore.Metadata, error) {
+	if len(sha256) < 7 {
```

Comment:
Make that a constant.

Add the returned errors in the comment.

Reply:
Also, considering the origin of the sha256 is a query string, I think we should have a regexp around this. This should ensure that the sha has valid characters. This would prevent any additional load on the database.

Reply:
Done